src/ExportHistory/Orchestrations/ExportJobOrchestrator.cs

```diff
@@ -0,0 +1,397 @@
+                    currentState.Config == null ||
+                    currentState.Status != ExportJobStatus.Active)
+                {
+                    logger.ExportJobOperationWarning(jobId, nameof(ExportJobOrchestrator), "Job is no longer active");
+                    return null;
+                }
+
+                // if (currentState.Checkpoint is not null &&
+                //     string.IsNullOrEmpty(currentState.Checkpoint.LastInstanceKey))
+                // {
+                //     if (config.Mode == ExportMode.Batch)
+                //     {
+                //         logger.ExportJobOperationInfo(jobId, nameof(ExportJobOrchestrator), "No more instances to export - export complete.");
+                //         break;
+                //     }
+                //     else if (config.Mode == ExportMode.Continuous)
+                //     {
+                //         logger.ExportJobOperationInfo(jobId, nameof(ExportJobOrchestrator), "No more instances to export currently - will check again later.");
+                //         await context.CreateTimer(ContinuousExportIdleDelay, default);
+                //         continue;
+                //     }
+                //     else
+                //     {
+                //         throw new InvalidOperationException("Invalid export mode.");
+                //     }
+                // }
+
+                // Call activity to list terminal instances with only necessary information
+                ListTerminalInstancesRequest listRequest = new ListTerminalInstancesRequest(
+                    CompletedTimeFrom: currentState.Config.Filter.CompletedTimeFrom,
+                    CompletedTimeTo: currentState.Config.Filter.CompletedTimeTo,
+                    RuntimeStatus: currentState.Config.Filter.RuntimeStatus,
+                    LastInstanceKey: currentState.Checkpoint?.LastInstanceKey,
+                    MaxInstancesPerBatch: currentState.Config.MaxInstancesPerBatch);
+
+                InstancePage pageResult = await context.CallActivityAsync<InstancePage>(
+                    nameof(ListTerminalInstancesActivity),
+                    listRequest);
+
+                List<string> instancesToExport = pageResult.InstanceIds;
+                long scannedCount = instancesToExport.Count;
+
+                if (scannedCount == 0)
+                {
+                    if (config.Mode == ExportMode.Continuous)
+                    {
+                        logger.ExportJobOperationInfo(jobId, nameof(ExportJobOrchestrator), "No more instances to export - export complete.");
+                        await context.CreateTimer(ContinuousExportIdleDelay, default);
+                        continue;
+                    }
+                    else if (config.Mode == ExportMode.Batch)
+                    {
+                        logger.ExportJobOperationInfo(jobId, nameof(ExportJobOrchestrator), "No more instances to export currently - will retry later.");
```

Comment:
This makes me wonder if we should force all batch operations to have an end date in the past so they can actually run and complete instead of waiting.